File: c/Euclidean_Algorithm.c
#include <stdio.h>
int euclidean(int a, int b){
	int mod;
	mod = a%b;
	if(!mod){
		return b;
	} else {
		return euclidean(b,mod);
	}
}

int main (void){
	int a, b;
	scanf("%d %d",&a,&b);
	if(a > b){
		printf("%d\n",euclidean(a,b));
	} else {
		printf("%d\n",euclidean(b,a));
	}
	return 0;
}

File: hs/Euclidean_Algorithm.hs
import Data.List

main :: IO ()
main = getLine >>= print . euclidean . sortBy (flip compare) . map read . words

euclidean :: [Integer] -> Integer
euclidean [a, b]
    | m == 0 = b
    | otherwise = euclidean [b, m]
    where
        m = a `mod` b
